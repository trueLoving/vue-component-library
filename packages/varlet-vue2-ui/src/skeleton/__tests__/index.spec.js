import example from '../example'
import Vue from 'vue'
import Skeleton from '..'
import VarSkeleton from '../Skeleton'
import { mount } from '@vue/test-utils'

test('test skeleton example', () => {
  const wrapper = mount(example)
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test skeleton plugin', () => {
  Vue.use(Skeleton)
  expect(Vue.component(Skeleton.name)).toBeTruthy()
})

test('test skeleton loading', async () => {
  const wrapper = mount(VarSkeleton, {
    propsData: {
      loading: true,
    },
    scopedSlots: {
      default: "<div class='skeleton-default-slot'></div>",
    },
  })

  expect(wrapper.find('.skeleton-default-slot').exists()).toBeFalsy()

  await wrapper.setProps({ loading: false })
  expect(wrapper.find('.skeleton-default-slot').exists()).toBeTruthy()

  wrapper.destroy()
})

test('test skeleton styles', () => {
  const wrapper = mount(VarSkeleton, {
    propsData: {
      title: true,
      card: true,
      avatar: true,
      titleWidth: '100px',
      cardHeight: '100px',
      avatarSize: '100px',
      rows: 2,
      rowsWidth: ['600px', '500px'],
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test skeleton fullscreen', () => {
  const wrapper = mount(VarSkeleton, {
    propsData: {
      fullscreen: true,
      fullscreenZIndex: 1000,
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})
