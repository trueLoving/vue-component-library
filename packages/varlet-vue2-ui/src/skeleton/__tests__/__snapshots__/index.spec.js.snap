// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test skeleton example 1`] = `
"<div class=\\"var-skeleton-example\\">
  <div class=\\"app-type\\">基本使用</div>
  <div class=\\"var--box var-skeleton\\">
    <!---->
    <div class=\\"var-skeleton__content\\">
      <!---->
      <div class=\\"var-skeleton__article\\">
        <!---->
        <div class=\\"var-skeleton__section\\">
          <!---->
          <div class=\\"var-skeleton__row\\">
            <div class=\\"var-skeleton--animation\\"></div>
          </div>
          <div class=\\"var-skeleton__row\\">
            <div class=\\"var-skeleton--animation\\"></div>
          </div>
          <div class=\\"var-skeleton__row\\">
            <div class=\\"var-skeleton--animation\\"></div>
          </div>
        </div>
      </div>
    </div>
    <!---->
  </div>
  <div class=\\"app-type\\">显示标题</div>
  <div class=\\"var--box var-skeleton\\">
    <!---->
    <div class=\\"var-skeleton__content\\">
      <!---->
      <div class=\\"var-skeleton__article\\">
        <!---->
        <div class=\\"var-skeleton__section\\">
          <div class=\\"var-skeleton__title\\">
            <div class=\\"var-skeleton--animation\\"></div>
          </div>
          <div class=\\"var-skeleton__row\\">
            <div class=\\"var-skeleton--animation\\"></div>
          </div>
          <div class=\\"var-skeleton__row\\">
            <div class=\\"var-skeleton--animation\\"></div>
          </div>
          <div class=\\"var-skeleton__row\\">
            <div class=\\"var-skeleton--animation\\"></div>
          </div>
        </div>
      </div>
    </div>
    <!---->
  </div>
  <div class=\\"app-type\\">自定义段落高度</div>
  <div class=\\"var--box var-skeleton\\">
    <!---->
    <div class=\\"var-skeleton__content\\">
      <!---->
      <div class=\\"var-skeleton__article\\">
        <!---->
        <div class=\\"var-skeleton__section\\">
          <div class=\\"var-skeleton__title\\">
            <div class=\\"var-skeleton--animation\\"></div>
          </div>
          <div class=\\"var-skeleton__row\\" style=\\"width: 200px;\\">
            <div class=\\"var-skeleton--animation\\"></div>
          </div>
          <div class=\\"var-skeleton__row\\" style=\\"width: 100px;\\">
            <div class=\\"var-skeleton--animation\\"></div>
          </div>
          <div class=\\"var-skeleton__row\\" style=\\"width: 50px;\\">
            <div class=\\"var-skeleton--animation\\"></div>
          </div>
        </div>
      </div>
    </div>
    <!---->
  </div>
  <div class=\\"app-type\\">显示头像</div>
  <div class=\\"var--box var-skeleton\\">
    <!---->
    <div class=\\"var-skeleton__content\\">
      <!---->
      <div class=\\"var-skeleton__article\\">
        <div class=\\"var-skeleton__avatar\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
        <div class=\\"var-skeleton__section\\">
          <div class=\\"var-skeleton__title\\">
            <div class=\\"var-skeleton--animation\\"></div>
          </div>
          <div class=\\"var-skeleton__row\\">
            <div class=\\"var-skeleton--animation\\"></div>
          </div>
          <div class=\\"var-skeleton__row\\">
            <div class=\\"var-skeleton--animation\\"></div>
          </div>
          <div class=\\"var-skeleton__row\\">
            <div class=\\"var-skeleton--animation\\"></div>
          </div>
        </div>
      </div>
    </div>
    <!---->
  </div>
  <div class=\\"app-type\\">显示卡片</div>
  <div class=\\"var--box var-skeleton\\">
    <!---->
    <div class=\\"var-skeleton__content\\">
      <div class=\\"var-skeleton__card\\">
        <div class=\\"var-skeleton--animation\\"></div>
      </div>
      <div class=\\"var-skeleton__article\\">
        <div class=\\"var-skeleton__avatar\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
        <div class=\\"var-skeleton__section\\">
          <div class=\\"var-skeleton__title\\">
            <div class=\\"var-skeleton--animation\\"></div>
          </div>
          <div class=\\"var-skeleton__row\\">
            <div class=\\"var-skeleton--animation\\"></div>
          </div>
          <div class=\\"var-skeleton__row\\">
            <div class=\\"var-skeleton--animation\\"></div>
          </div>
          <div class=\\"var-skeleton__row\\">
            <div class=\\"var-skeleton--animation\\"></div>
          </div>
        </div>
      </div>
    </div>
    <!---->
  </div>
  <div class=\\"app-type\\">全屏模式</div> <button class=\\"var-button var--box button var-button--normal var--inline-flex var-button--danger var-elevation--2\\">
    <!---->
    <div class=\\"var-button__content\\">
      切换全屏模式
    </div>
  </button>
  <div class=\\"var--box var-skeleton\\">
    <div class=\\"var-skeleton__data\\"></div>
    <!---->
    <!---->
  </div>
</div>"
`;

exports[`test skeleton fullscreen 1`] = `
"<div class=\\"var--box var-skeleton\\">
  <!---->
  <!---->
  <div class=\\"var-skeleton__fullscreen\\" style=\\"z-index: 1000;\\">
    <div class=\\"var-skeleton--animation\\"></div>
  </div>
</div>"
`;

exports[`test skeleton styles 1`] = `
"<div class=\\"var--box var-skeleton\\">
  <!---->
  <div class=\\"var-skeleton__content\\">
    <div class=\\"var-skeleton__card\\" style=\\"height: 100px;\\">
      <div class=\\"var-skeleton--animation\\"></div>
    </div>
    <div class=\\"var-skeleton__article\\">
      <div class=\\"var-skeleton__avatar\\" style=\\"width: 100px; height: 100px;\\">
        <div class=\\"var-skeleton--animation\\"></div>
      </div>
      <div class=\\"var-skeleton__section\\">
        <div class=\\"var-skeleton__title\\" style=\\"width: 100px;\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
        <div class=\\"var-skeleton__row\\" style=\\"width: 600px;\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
        <div class=\\"var-skeleton__row\\" style=\\"width: 500px;\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
      </div>
    </div>
  </div>
  <!---->
</div>"
`;
